=== modified file 'src/main.cpp'
--- compiz-0.9.12.1.orig/src/main.cpp	2013-02-21 17:05:46 +0000
+++ compiz-0.9.12.1/src/main.cpp	2015-04-23 20:18:28 +0000
@@ -38,6 +38,11 @@
 #include "privatescreen.h"
 #include "privatestackdebugger.h"
 
+namespace
+{
+    bool autoAddCCP = true;
+}
+
 void
 CompManager::usage ()
 {
@@ -48,6 +53,7 @@
             "  --sm-disable          Disable session management\n"
             "  --sm-client-id ID     Session management client ID\n"
             "  --keep-desktop-hints  Retain existing desktop hints\n"
+            "  --no-auto-add-ccp     Do not automatically load the ccp plugin\n"
             "  --sync                Make all X calls synchronous\n"
             "  --debug               Enable debug mode\n"
             "  --version             Show the program version\n"
@@ -99,6 +105,11 @@
 	{
 	    useDesktopHints = true;
 	}
+	else if (!strcmp (argv[i], "--no-auto-add-ccp"))
+	{
+	    compLogMessage ("core", CompLogLevelInfo, "running without the ccp plugin, no settings will be loaded");
+	    autoAddCCP = false;
+	}
 	else if (!strcmp (argv[i], "--replace"))
 	{
 	    replaceCurrentWm = true;
@@ -151,6 +162,9 @@
 
     modHandler = new ModifierHandler ();
 
+    if (initialPlugins.empty () && autoAddCCP)
+        initialPlugins.push_back ("ccp");
+
     if (!screen->init (displayName))
 	return false;
 

=== modified file 'tests/xorg-gtest/src/compiz-xorg-gtest.cpp'
--- compiz-0.9.12.1.orig/tests/xorg-gtest/src/compiz-xorg-gtest.cpp	2013-07-18 13:06:24 +0000
+++ compiz-0.9.12.1/tests/xorg-gtest/src/compiz-xorg-gtest.cpp	2015-04-23 20:18:28 +0000
@@ -778,6 +778,7 @@
     if (flags & ct::CompizProcess::ReplaceCurrentWM)
 	args.push_back ("--replace");
 
+    args.push_back ("--no-auto-add-ccp");
     args.push_back ("--send-startup-message");
 
     /* Copy in plugin list and set environment variables */

